src/auth.rs

```diff
@@ -17,6 +22,57 @@ pub fn parse_auth(auth: String, host: &str) -> io::Result<(String, Option<String
     }
 }
 
+fn netrc_path() -> Option<PathBuf> {
+    match env::var("NETRC") {
```

Comment:
It's slightly safer to do it like this:
```suggestion
    match env::var_os("NETRC") {
```
That way we won't panic if there's invalid unicode in the variable. You'll get an `OsString` instead of a `String`, but that can be freely converted to `PathBuf`.